pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -238,6 +238,16 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 }
 
 func getServerByName(client *gophercloud.ServiceClient, name string) (*servers.Server, error) {
+	// try to read from local file (cloud-init) or metadata server first
+	id, err := readInstanceID()
+	if err == nil {
+		server, err := servers.Get(client, id).Extract()
+		if err == nil {
+			return server, nil
```

Comment:
This ignores the `name` argument and assumes it's the name of the local node.  I feel that's too risky.

Perhaps we could move this logic further up the call stack to somewhere were we know we're looking for the local node and can avoid using `getServerByName` altogether?